examples/siro_sandbox/app_states/app_state_socialnav_study.py

```diff
@@ -0,0 +1,459 @@
+from app_states.app_state_abc import AppState
+from camera_helper import CameraHelper
+from controllers.fetch_baselines_controller import FetchState
+from controllers.gui_controller import GuiHumanoidController
+from gui_navigation_helper import GuiNavigationHelper
+from hablab_utils import get_agent_art_obj_transform, get_grasped_objects_idxs
+
+from habitat.gui.gui_input import GuiInput
+from habitat.gui.text_drawer import TextOnScreenAlignment
+
+
+class AppStateSocialNavStudy(AppState):
+    def __init__(
+        self,
+        sandbox_service,
+        gui_agent_ctrl,
+        robot_agent_ctrl,
+    ):
+        self._sandbox_service = sandbox_service
+        self._gui_agent_ctrl = gui_agent_ctrl
+        self._state_machine_robot_ctrl = robot_agent_ctrl
+
+        # cache items from config; config is expensive to access at runtime
+        config = self._sandbox_service.config
+        self._end_on_success = config.habitat.task.end_on_success
+        self._obj_succ_thresh = config.habitat.task.obj_succ_thresh
+        self._success_measure_name = config.habitat.task.success_measure
+
+        self._can_grasp_place_threshold = (
+            self._sandbox_service.args.can_grasp_place_threshold
+        )
+
+        self._cam_transform = None
+
+        self._held_target_obj_idx = None
+        self._num_remaining_objects = None  # resting, not at goal location yet
+        self._num_busy_objects = None  # currently held by non-gui agents
+
+        # will be set in on_environment_reset
+        self._target_obj_ids = None
+        self._goal_positions = None
+
+        self._camera_helper = CameraHelper(
+            self._sandbox_service.args, self._sandbox_service.gui_input
+        )
+        self._first_person_mode = self._sandbox_service.args.first_person_mode
+
+        self._nav_helper = GuiNavigationHelper(
+            self._sandbox_service, self.get_gui_controlled_agent_index()
+        )
+        self._episode_helper = self._sandbox_service.episode_helper
+
+    def on_environment_reset(self, episode_recorder_dict):
+        self._held_target_obj_idx = None
+        self._num_remaining_objects = None  # resting, not at goal location yet
+        self._num_busy_objects = None  # currently held by non-gui agents
+
+        sim = self.get_sim()
+        temp_ids, goal_positions_np = sim.get_targets()
+        self._target_obj_ids = [
+            sim._scene_obj_ids[temp_id] for temp_id in temp_ids
+        ]
+        self._goal_positions = [mn.Vector3(pos) for pos in goal_positions_np]
+
+        self._nav_helper.on_environment_reset()
+
+        self._camera_helper.update(self._get_camera_lookat_pos(), dt=0)
+
+        if episode_recorder_dict:
+            episode_recorder_dict["target_obj_ids"] = self._target_obj_ids
+            episode_recorder_dict["goal_positions"] = self._goal_positions
+
+    def get_sim(self):
+        return self._sandbox_service.sim
+
+    def _update_grasping_and_set_act_hints(self):
```

Comment:
If you're feeling motivated, I can imagine a `GuiRearrangeHelper` class that encapsulates this grasping logic and visualization and gets shared between AppStateRearrange and AppStateSocialNavStudy.

However I'm also happy to defer this to a future PR. I'm curious to see if the specifics here end up diverging a bit between the two app states over time.

Reply:
Will work on this in separate PR.